src/dms/core/EdFi.DataManagementService.Core/Middleware/ProvideEducationOrganizationHierarchyMiddleware.cs

```diff
@@ -93,8 +93,14 @@ private long ExtractEducationOrganizationId(PipelineContext context)
         ];
 
         return context
-            .DocumentSecurityElements.EducationOrganization.Where(e => parentTypes.Contains(e.ResourceName))
-            .Select(e => e.Id.Value)
+            .DocumentSecurityElements.EducationOrganization
+            .Where(e => parentTypes.Contains(e.ResourceName))
+            .Select(e =>
+            {
+                return long.TryParse(e.Id.Value, out var idValue) ? (long?)idValue : null;
```

Comment:
If we are changing from long to string, these methods should be returning strings. There should be almost no  variable, parameter or return types of `long` for edorgids.

Reply:
I updated it . Thanks !